### PR TITLE
#1704 create directions table general look 2

### PR DESCRIPTION
**Pull request by nazarstig**

### CODE REVIEW

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  ngOnInit(): void {
+    this.store.dispatch([new GetFieldDescriptionByInstitutionId(this.institution.id),
+      new GetAllInstitutionsHierarchy()]);
+
+    this.institutionFieldDesc$.pipe(
+      filter((institutionFieldDesc: InstitutionFieldDescription[]) => !!institutionFieldDesc),
+      distinctUntilChanged(),
+      takeUntil(this.destroy$)
+    ).subscribe((institutionFieldDesc: InstitutionFieldDescription[]) => {
+        this.store.snapshot
```
Comment (litvinets) [inline src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts:42]: probably should be removed

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        this.displayedColumns = institutionFieldDesc.map((ins: InstitutionFieldDescription) => ins.title);
+      });
+    this.institutionsHierarchies$.pipe(
+      filter((institutiionHierarchies: InstituitionHierarchy[]) => !!institutiionHierarchies),
+      distinctUntilChanged(),
+      map((institutionHierarchies: InstituitionHierarchy[]) =>
+        this.createDirectionTableRecords(institutionHierarchies.filter(i => i.institution.title == this.institution.title))
```
Comment (litvinets) [inline src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts:49]: this.createDirectionTableRecords(institutionHierarchies.filter(i => i.institution.title == this.institution.title))
```suggestion
        this.createDirectionTableRecords(institutionHierarchies.filter((ins: InstituitionHierarchy) => ins.institution.title === this.institution.title))
```

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      this.isLoaded = true;
+      this.dataSource = new MatTableDataSource(this.records)
+    });
+  }
+
+  private createDirectionTableRecords(institutionalHierarchies: InstituitionHierarchy[]) {
+    if (institutionalHierarchies) {
+      this.institutionalHierarchies = institutionalHierarchies;
+      this.records = [];
+      const firstLevelInstitutions = this.institutionalHierarchies.filter((ins: InstituitionHierarchy) => ins.hierarchyLevel == 1);
```
Comment (litvinets) [inline src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts:62]: ===

**Review by litvinets — CHANGES_REQUESTED** [review]

File: src/app/shared/services/institutions/institutions.service.ts [hunk 4]
```diff
@@ -13,6 +13,10 @@ export class InstitutionsService {
     return this.http.get<Institution[]>('/api/v1/Institution/GetAll');
   }
 
+  getAllInstitutionHierarchies(): Observable<InstituitionHierarchy[]> {
```
Comment (Y-Taras) [inline src/app/shared/services/institutions/institutions.service.ts:16]: Annotations would not hurt

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.scss [hunk 5]
```diff
@@ -0,0 +1 @@
+@import "src/app/shared/styles/tables.scss";
```
Comment (Y-Taras) [inline src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.scss:1]: please use relative import if it possible

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  dataSource: MatTableDataSource<object> = new MatTableDataSource([{}]);
+
+  constructor(private store: Store) {
+  }
+
+  ngOnInit(): void {
+    this.store.dispatch([new GetFieldDescriptionByInstitutionId(this.institution.id),
+      new GetAllInstitutionsHierarchy()]);
+
+    this.institutionFieldDesc$.pipe(
```
Comment (Y-Taras) [inline src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts:37]: How about create separate method for all that stuff ( I mean it is not 7 line of code nad a little bit too much for ngoninit)

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts (hunk 2 above)
Comment (Y-Taras) [reply src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.ts:49]: yep it is not good to use  not strict equal

File: src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.scss (hunk 5 above)
Comment (nazarstig) [reply src/app/shell/admin-tools/data/directions-wrapper/directions-institution-hierarchies-list/directions-institution-hierarchies-list.component.scss:1]: with styles should be as it is

**Review by litvinets — APPROVED** [review]

**Pull request merged**